packages/orchestration/test/cosmos-interchain-service.test.ts

```diff
@@ -0,0 +1,397 @@
+  t.is(localAddr, localAddr2, 'provideICQConnection is idempotent');
+
+  const [result] = await E(icqConnection).query([
+    toRequestQueryJson(
+      QueryBalanceRequest.toProtoMsg({
+        address: 'cosmos1test',
+        denom: 'uatom',
+      }),
+    ),
+  ]);
+
+  t.like(QueryBalanceResponse.decode(decodeBase64(result.key)), {
+    balance: { amount: '0', denom: 'uatom' },
+  });
+
+  const icqConnection3 = await E(cosmosInterchainService).provideICQConnection(
+    CONTROLLER_CONNECTION_ID,
+    'icq-2',
+  );
+  const localAddr3 = await E(icqConnection3).getLocalAddress();
+  t.not(
+    localAddr3,
+    localAddr2,
+    'non default version results in a new connection',
+  );
+
+  const icqConnection4 = await E(cosmosInterchainService).provideICQConnection(
+    CONTROLLER_CONNECTION_ID,
+    'icq-2',
+  );
+  const localAddr4 = await E(icqConnection4).getLocalAddress();
+  t.is(localAddr3, localAddr4, 'custom version is idempotent');
+
+  const icqConnection5 = await E(cosmosInterchainService).provideICQConnection(
+    'connection-99',
+  );
+  const localAddr5 = await E(icqConnection5).getLocalAddress();
+
+  const getPortId = (lAddr: LocalIbcAddress) => lAddr.split('/')[2];
+  const uniquePortIds = new Set(
+    [localAddr, localAddr2, localAddr3, localAddr4, localAddr5].map(getPortId),
+  );
+  t.regex([...uniquePortIds][0], /icqcontroller-\d+/);
+  t.is(uniquePortIds.size, 1, 'all connections share same port');
+
+  await t.throwsAsync(
+    // @ts-expect-error icqConnectionKit doesn't have a port method
+    () => E(icqConnection).getPort(),
+    undefined,
+    'ICQConnection Kit does not expose its port',
+  );
+});
+
+test.serial('makeAccount returns an IcaAccountKit', async t => {
+  const {
+    bootstrap: { cosmosInterchainService },
+  } = await commonSetup(t);
+
+  const account = await E(cosmosInterchainService).makeAccount(
+    CHAIN_ID,
+    HOST_CONNECTION_ID,
+    CONTROLLER_CONNECTION_ID,
+  );
+  const [localAddr, remoteAddr, chainAddr, port] = await Promise.all([
+    E(account).getLocalAddress(),
+    E(account).getRemoteAddress(),
+    E(account).getAddress(),
+    E(account).getPort(),
+  ]);
+  t.log(account, {
+    localAddr,
+    remoteAddr,
+    chainAddr,
+  });
+  t.regex(localAddr, /ibc-port\/icacontroller-\d+/);
+  t.regex(
+    remoteAddr,
+    new RegExp(`/ibc-hop/${CONTROLLER_CONNECTION_ID}`),
+    'remote address contains provided connectionId',
+  );
+  t.regex(
+    remoteAddr,
+    /icahost\/ordered/,
+    'remote address contains icahost port, ordered ordering',
+  );
+  t.regex(
+    remoteAddr,
+    /"version":"ics27-1"(.*)"encoding":"proto3"/,
+    'remote address contains version and encoding in version string',
+  );
+  t.true(
+    (
+      ['addListener', 'removeListener', 'connect', 'revoke'] as (keyof Port)[]
+    ).every(method => getMethodNames(port).includes(method)),
+    'IcaAccountKit returns a Port remotable',
+  );
```

Comment:
would be nice to see what's missing
```suggestion
  t.deepEqual(
    getMethodNames(port).sort(),
    ['addListener', 'connect', 'removeListener', 'revoke'] as (keyof Port)[],
    'IcaAccountKit returns a Port remotable',
  );
```